### ISSUE TITLE
Multiple obstacles?

### ISSUE THREAD

**Issue by jkurtz678** [Scrawk/2D-Fluid-Simulation#1]
I'm having trouble adding multiple obstacles to the scene. I've tried modifying the code in AddObstacles() as well as creating new obstacle textures/materials, but only one obstacle is ever appears in the scene. Would the solution be to modify the Obstacle shader, or am I missing something obvious? Thanks, this is a very cool project.

**Comment by Scrawk** [comment]
The add obstacle shader will overwrite anything in the texture when ran. That’s why you only get the last obstacle added. You need to add a second obstacle texture and use the read/write method like the other shader use to include what was previously in the texture.

Alternatively you could try changing the shader blend mode to additive.

**Comment by jkurtz678** [comment]
I appreciate the quick response! I think I understand the issue now. I found a temporary solution which involved adding additional point variables to the obstacle shader code ( uniform float2 _Point2; ) but I'm guessing that isn't the best way to do it. Your solutions seem better, I'll give them a shot. 

You might enjoy this animation I made, thanks again. 
![doorways](https://user-images.githubusercontent.com/29451475/69012253-9e748680-0928-11ea-9bb6-a29a4b10c21b.gif)
